src/fmu/dataio/types.py

```diff
@@ -1,10 +1,101 @@
 from __future__ import annotations
 
-from typing import Dict, Union
+import pathlib
+from typing import TYPE_CHECKING, Literal, MutableMapping, Union
 
 from typing_extensions import Annotated, TypeAlias
 
+if TYPE_CHECKING:
+    from pandas import DataFrame
+    from pyarrow import Table
+    from xtgeo.cube import Cube
+    from xtgeo.grid3d import Grid, GridProperty
+    from xtgeo.surface import RegularSurface
+    from xtgeo.xyz import Points, Polygons
+
+    # Local proxies due to xtgeo at the time of writing
+    # not have stubs/marked itself as a typed library.
+    # Ref.: https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-library-stubs-or-py-typed-marker
+    class CubeProxy(Cube):
+        ...
+
+    class GridProxy(Grid):
+        ...
+
+    class GridPropertyProxy(GridProperty):
+        ...
+
+    class RegularSurfaceProxy(RegularSurface):
+        ...
+
+    class PointsProxy(Points):
+        ...
+
+    class PolygonsProxy(Polygons):
+        ...
+
+    Sniffable: TypeAlias = Annotated[
```

Comment:
Suggestions for other names are very welcome.

Reply:
I nominate `Inferrable` or `TypeInferrable`

Deducible could work too, or `AutoAnnotatable`/`AutoAnnotated`